### PR TITLE
Realm Location Saving/Migration/Export

### PR DESCRIPTION
**Pull request by troach-sf**

### CODE REVIEW

File: android/app/src/main/java/org/pathcheck/covidsafepaths/storage/Location.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import io.realm.RealmObject
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.WritableNativeMap
+import io.realm.annotations.PrimaryKey
+import java.lang.Exception
+
+/*
+  Realm requires a no-op constructor. Need to use var and fill will default value
+ */
+open class Location(
```
Comment (qingqingliu) [inline android/app/src/main/java/org/pathcheck/covidsafepaths/storage/Location.kt:16]: so this has to be declared as open class?

File: android/app/src/main/java/org/pathcheck/covidsafepaths/storage/Location.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        return Location(
+            time = time,
+            latitude = latitude,
+            longitude = longitude,
+            source = source
+        )
+      } catch (exception: Exception) {
+        // possible react type-safe issues here
+        null
```
Comment (qingqingliu) [inline android/app/src/main/java/org/pathcheck/covidsafepaths/storage/Location.kt:89]: should we log error in some given way to track error?

File: ios/Location.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    location.speed.value = backgroundLocation.speed.floatValue
+    location.accuracy.value = backgroundLocation.accuracy.floatValue
+    location.altitudeAccuracy.value = backgroundLocation.altitudeAccuracy.floatValue
+    location.bearing.value = backgroundLocation.heading.floatValue
+    location.source = SOURCE_DEVICE
+    return location;
+  }
+  
+  static func fromImportLocation(dictionary: NSDictionary?, source: Int) -> Location? {
+    if dictionary == nil { return nil }
```
Comment (qingqingliu) [inline ios/Location.swift]: we can use guard let dictionary = dictionary else { return nil} here, that way, you can directly use dictionary below without "?"

File: ios/RealmWrapper.swift [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  @objc static let shared = RealmWrapper()
+  
+  private static let MINIMUM_TIME_INTERVAL = 60 * 4
+  private static let DAYS_TO_KEEP = 14
+  private static let KEY_LAST_SAVED_TIME = "lastSavedTime"
+  
+  let realmConfig: Realm.Configuration
+  
+  private override init() {
+    realmConfig = Realm.Configuration(objectTypes: [Location.self])
```
Comment (qingqingliu) [inline ios/RealmWrapper.swift]: we should call super.init() here

**Comment by qingqingliu** [comment]
shouldn't this PR go ahead self-reporting branch?

File: ios/RealmWrapper.swift (hunk 4 above)
Comment (troach-sf) [reply ios/RealmWrapper.swift]: Pushing up encryption work shortly which will remove this override.

File: android/app/src/main/java/org/pathcheck/covidsafepaths/storage/Location.kt (hunk 1 above)
Comment (troach-sf) [reply android/app/src/main/java/org/pathcheck/covidsafepaths/storage/Location.kt:16]: Yes, sounds like realm does something to inherit from this class so we need to leave it open.

**Review by qingqingliu — APPROVED** [review]
looks good

**Pull request merged**